src/gto-cas-login.php

```diff
@@ -531,11 +532,16 @@ function wp_validate_auth_cookie( $cookie = '', $scheme = '' ) {
       $user = \GlobalTechnology\CentralAuthenticationService\CASLogin::singleton()->get_user_by_guid( $guid );
 
       // return the user id or false if the user doesn't exist
-      return is_null( $user ) ? false : $user->ID;
+      $logged_in = is_null( $user ) ? false : $user->ID;
     }
 
-    // not a valid CAS session, so return false
-    return false;
+    if ( false === $logged_in ) {
+      setcookie( 'logged_in_cookie', ' ', time() - YEAR_IN_SECONDS, COOKIEPATH, COOKIE_DOMAIN );
+    } else {
+      setcookie( 'logged_in_cookie', 'logged_in', time() + 2 * HOUR_IN_SECONDS, COOKIEPATH, COOKIE_DOMAIN, true, true );
```

Comment:
Seems like the duration of this cookie should match whatever the session timeout of wp-admin session is. Is that 2 hours?

Reply:
We completely bypass WordPress sessions. Their default is 2 weeks. This checks if we have a valid CAS session, if so, it adds the cookie with 2 hour expiration. We have no access to when CAS sessions expire. This gets updated with each subsequent request to WordPress where the user is logged in, or expired if they have been logged out of CAS. There's a change this could exist for 2 hours beyond the CAS session, but It'll get removed on the first request for which we detect CAS session has expired.

Reply:
The concern is that if someone doesn't have this cookie, WP rocket will cache what they see. If they load a page while logged into wp-admin, it'll cache with that black admin nav bar at the top of the page. It's safer to make this cookie too long than too short if all we're using it for is to decide whether or not to cache.

Reply:
This is a brand new cookie, WPRocket wouldn't be aware of it. It should internally be using get_current_user() to determine if a user is logged in. This cookie was for nginx to bypass the cache, correct?